compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/BIRPackageSymbolEnter.java

```diff
@@ -477,14 +501,34 @@ private void defineErrorConstructor(Scope scope, BTypeSymbol typeDefSymbol) {
     private BType readBType(DataInputStream dataInStream) throws IOException {
         int typeCpIndex = dataInStream.readInt();
         CPEntry cpEntry = this.env.constantPool[typeCpIndex];
+        BType type = null;
         if (cpEntry != null) {
-            return ((CPEntry.ShapeCPEntry) cpEntry).shape;
+            type = ((CPEntry.ShapeCPEntry) cpEntry).shape;
+            if (type.tag != TypeTags.INVOKABLE) {
+                return type;
+            }
+        }
+        if (type == null) {
+            byte[] e = env.unparsedBTypeCPs.get(typeCpIndex);
+            type = new BIRTypeReader(new DataInputStream(new ByteArrayInputStream(e))).readType(typeCpIndex);
+            addShapeCP(type, typeCpIndex);
+        }
+
+        if (type.tag == TypeTags.INVOKABLE) {
+            return createClonedInvokableTypeWithTsymbol((BInvokableType) type);
         }
-        byte[] e = env.unparsedBTypeCPs.get(typeCpIndex);
-        BType bType = new BIRTypeReader(new DataInputStream(new ByteArrayInputStream(e))).readType(typeCpIndex);
-        addShapeCP(bType, typeCpIndex);
 
-        return bType;
+        return type;
+    }
+
+    private BInvokableType createClonedInvokableTypeWithTsymbol(BInvokableType bInvokableType) {
+        BInvokableType clonedType = new BInvokableType(bInvokableType.paramTypes, bInvokableType.restType,
+                bInvokableType.retType, null);
+        clonedType.tsymbol = Symbols.createInvokableTypeSymbol(SymTag.FUNCTION_TYPE,
+                bInvokableType.flags, env.pkgSymbol.pkgID, null,
+                env.pkgSymbol.owner);
+        //TODO: tsymbol param values should be read from bir and added here
```

Comment:
Do we need this TODO here? Can we have an issue to track this. There are several places with TODOs

Reply:
This is related to a future enhancement in function pointer (supporting default parameters). There is already an issue https://github.com/ballerina-platform/ballerina-lang/issues/19542  to track that. 

This was more of a hint for the developer who is going to implement that feature on where to begin. I've seen these TODOS everywhere in the repo, so I assumed it was okay. 

I have removed that.

Reply:
IMO, TODOs are ok as long as we have an issue to track the requried future changes and that issue number is also included in the todo comment.